bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlSequence.java

```diff
@@ -626,17 +625,12 @@ public Object next() {
     }
 
     private Type getSequenceType(Type tempExprType) {
-        switch (tempExprType.getTag()) {
-            case TypeTags.XML_ELEMENT_TAG:
-                return new BXmlType(PredefinedTypes.TYPE_ELEMENT, false);
-            case TypeTags.XML_COMMENT_TAG:
-                return new BXmlType(PredefinedTypes.TYPE_COMMENT, false);
-            case TypeTags.XML_PI_TAG:
-                return new BXmlType(PredefinedTypes.TYPE_PROCESSING_INSTRUCTION, false);
-            default:
-                // Since 'xml:Text is same as xml<'xml:Text>
-                return PredefinedTypes.TYPE_TEXT;
-        }
+        return switch (tempExprType.getTag()) {
+            case TypeTags.XML_ELEMENT_TAG -> PredefinedTypes.XML_ELEMENT_SEQUENCE_TYPE;
+            case TypeTags.XML_COMMENT_TAG -> PredefinedTypes.XML_COMMENT_SEQUENCE_TYPE;
+            case TypeTags.XML_PI_TAG -> PredefinedTypes.XML_PI_SEQUENCE_TYPE;
+            default -> PredefinedTypes.XML_TEXT_SEQUENCE_TYPE;
+        };
```

Comment:
Have we verified this with the spec? AFAIK `xml<xml:Text>` is equivalent to `xml:Text` but other xml singletons are not the same. Here the method `getSequenceType ` expected to return a xml sequence which belongs to type `xml`.

Reply:
As per the offline discussion, we can go ahead with this approach.